apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/SizedBufferingConsumer.cs

```diff
@@ -69,61 +64,25 @@ public static SizedBufferingConsumer<T> Create(IConsumer<T> consumer, Func<T, in
         public override void Dispose() => Flush();
 
         /// <inheritdoc />
-        protected override void ReceiveWithSemaphoreHeld(IEnumerable<T> items)
+        protected override bool ReceiveWithSemaphoreHeld(IEnumerable<T> items)
```

Comment:
Given the return type change, I wonder if a `Try` prefix might make more sense? Although, it's not necessarily clear what `true` means without looking at the code.

Reply:
I agree that after the return type change the method name is not the best so I'm super open to a better name, but I wouldn't use `Try`, in C# using the `Try` prefix in a method returning `bool` is more associated with the returned value indicating whether the operation was succesfull or not, which is not the case, since the receiving will always be succesfull (unless exception) and the returning value in this case indicates whether the caller needs to flush the buffer or not. I'm not super worried about this because this is a `protected` method inside an `internal` class, and the meaning of the returned value is already indicated in docs. That said, if a better name comes up, happy to use it.
Oh, and also, I've been wanting to refactor (basically decoupling the buffering strategy from receiving and flushing) some of this for a while now, it probably won't happen soon, but it'll happen at some point.